sdk/keyvault/internal/challenge_policy.go

```diff
@@ -57,6 +58,10 @@ func (k *KeyVaultChallengePolicy) Do(req *policy.Request) (*http.Response, error
 			return nil, err
 		}
 
+		if resp.StatusCode > 399 {
+			// the request failed, don't try any further
+			return resp, nil
+		}
```

Comment:
This `resp` is nearly always going to fail, because it's a challenge auth (will return a 401). This should drop down until after the `err = k.findScopeAndTenant(resp)`.

Reply:
Fixed by skipping the 401.